### PR TITLE
Switch start and end date if end date is before start date

### PR DESCRIPTION
**Pull request by jonyardley**
As there is no validation in contentful if the `startTime` is after the `endTime` then we should switch them around as to not break elements in the UI.

I have also moved the locale into a separate module. (Probably should have done this in a separate PR but thought some housekeeping wouldn't go amiss).

Fixes: #73

## Pre-flight check-list
* [x] Documentation (N/A)
* [x] Unit tests

## Pre-merge check-list

* [ ] Tester approved

## TestPlan
- [ ] Create/change an event in contentful with the start date before the end date.
- [ ] App does not break and start and end date should be switched.

### CODE REVIEW

**Comment by declanslevin** [comment]
So the app isn't crashing when an event from Contentful has reversed start/end times, however this event's times aren't being corrected, instead the times are being flipped for all the other events (with correct start/end times):

screenshot_20180327-114409

"Swapped Date Test Event" is the only event in the list that has reversed dates. 

Tested on a Galaxy S8

File: src/integrations/storage.js [hunk 1]
```diff
@@ -32,12 +34,38 @@ const addOrUpdateEntry = (entries, newEntry) => {
...
+  if (!entry.fields) {
+    return entry;
+  }
+
+  const { startTime, endTime } = entry.fields;
+
+  if (
+    startTime !== undefined &&
+    endTime !== undefined &&
+    isBefore(endTime[locale], startTime[locale])
```
Comment (joaquimadraz) [inline src/integrations/storage.js]: Shouldn't this be the other way around? If the `startTime` is before `endTime` no need to switch and return earlier right?

**Comment by jonyardley** [comment]
I can pick this up. My bad sozza

File: src/integrations/storage.test.js [hunk 2]
```diff
@@ -201,3 +201,49 @@ describe("loadCmsData", () => {
...
+
+describe("correctDates", () => {
+  it("switches dates if end date is before start date", async () => {
+    const cmsData = {
+      entries: [
+        {
+          sys: { id: "1" },
+          fields: {
+            startTime: { "en-GB": "2018-07-07T17:30+01:00" },
+            endTime: { "en-GB": "2018-07-07T18:30+01:00" }
```
[the remark this answers is not in the document]
Comment (jonyardley) [reply src/integrations/storage.test.js]: 🎺

**Comment by jonyardley** [comment]
@declanslevin this is fixed now :)

**Comment by jonyardley** [comment]
screen shot 2018-04-08 at 11 53 49

**Comment by jonyardley** [comment]
closing in favour of https://github.com/redbadger/pride-london-app/pull/106

**Pull request closed without merging**